From 67a991a89e5b28dabf8372017b2b33e67585e6c4 Mon Sep 17 00:00:00 2001
From: Eric W. Biederman <ebiederm@xmission.com>
Date: Sat, 18 Jun 2011 06:33:29 -0700
Subject: [PATCH 05/16] proc: Add a test for proc namespace inodes

Implement a test to see if an inode is a proc
namespace inode.

Signed-off-by: Eric W. Biederman <ebiederm@xmission.com>
---
 fs/proc/namespaces.c    |    4 ++++
 include/linux/proc_fs.h |    6 ++++++
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/fs/proc/namespaces.c b/fs/proc/namespaces.c
index 041b64f..fd3ca05 100644
--- a/fs/proc/namespaces.c
+++ b/fs/proc/namespaces.c
@@ -330,3 +330,7 @@ out_invalid:
 	return ERR_PTR(-EINVAL);
 }
 
+bool proc_ns_inode(struct inode *inode)
+{
+	return inode->i_fop == &ns_file_operations;
+}
diff --git a/include/linux/proc_fs.h b/include/linux/proc_fs.h
index 5ebb8d7..2dfe699 100644
--- a/include/linux/proc_fs.h
+++ b/include/linux/proc_fs.h
@@ -177,6 +177,7 @@ extern struct proc_dir_entry *proc_net_mkdir(struct net *net, const char *name,
 	struct proc_dir_entry *parent);
 
 extern struct file *proc_ns_fget(int fd);
+extern bool proc_ns_inode(struct inode *inode);
 
 extern int proc_alloc_inum(unsigned int *pino);
 extern void proc_free_inum(unsigned int inum);
@@ -234,6 +235,11 @@ static inline struct file *proc_ns_fget(int fd)
 	return ERR_PTR(-EINVAL);
 }
 
+static bool proc_ns_inode(struct inode *inode)
+{
+	return false;
+}
+
 static inline int proc_alloc_inum(unsigned int *inum)
 {
 	*inum = 1;
-- 
1.7.4.1

